### PR TITLE
Set PyPI publish job with Github Actions

### PR DESCRIPTION
**Pull request by volpatto**

### CODE REVIEW

File: .dependencies/python.devenv.yml [hunk 1]
```diff
@@ -7,4 +7,4 @@ dependencies:
   - tabulate
   - colorama
   - python={{ python_version }}
-  - twine
+  - setuptools<49.0
```
Comment (volpatto) [inline .dependencies/python.devenv.yml:10]: Remark: for some reason, the package build with `setuptools>=49.0` breaks for some reason, maybe due to a new Python packaging standard. For now, this version constraint is enough.

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+jobs:
+  publish:
```
Comment (volpatto) [inline .github/workflows/pypi.yml:11]: This is the job that publishes Reaktoro to PyPI. It runs only when a tagged commit is pushed to master.

File: .dependencies/python.devenv.yml (hunk 1 above)
Comment (volpatto) [reply .dependencies/python.devenv.yml:10]: This does not work for Windows. Pinning to 49.5.0.

File: ci/check_compiled_files.py [hunk 3]
```diff
@@ -14,7 +14,13 @@
...
 
 if sys.platform.startswith('linux'):
-    assert (artifacts / 'lib/libReaktoro.so').is_file()
+    if (artifacts / 'lib64').is_dir():
+        assert (artifacts / 'lib64/libReaktoro.so').is_file()
+    elif (artifacts / 'lib').is_dir():
+        assert (artifacts / 'lib/libReaktoro.so').is_file()
+    else:
+        raise RuntimeError(f"Both {artifacts / 'lib64/libReaktoro.so'} " 
+            f"and {artifacts / 'lib/libReaktoro.so'} are not found.")
```
Comment (volpatto) [inline ci/check_compiled_files.py:23]: CC @tadeu. For some reason, in my Manjaro I don't have `lib`, but I have `lib64` instead. This is a workaround that we can improve afterwards. I'm open for suggestions!

Comment (allanleal) [reply ci/check_compiled_files.py:23]: I think in the conda-forge feedstock of autodiff, I had to set `CMAKE_LIBRARY_OUTPUT_DIRECTORY` to `lib` to ensure no `lib64` is used.

**Pull request merged**